openai-api-integration/README.md

```diff
@@ -0,0 +1,139 @@
+# OpenAI API Integration Example
+
+This project demonstrates how to integrate OpenAI API with Quex data oracles to fetch AI-generated responses on-chain. The example sends a sentiment analysis prompt to OpenAI's chat completions API and stores the numeric sentiment score (0-100) in a Solidity contract.
+
+## Overview
+
+The `OpenAIIntegration` contract:
+- Sends POST requests to OpenAI's chat completions endpoint
+- Uses encrypted private headers to securely pass the API key
+- Extracts sentiment scores from OpenAI responses
+- Stores the latest sentiment score on-chain
+- Emits events when new responses are received
+
+## Set Up
+
+Ensure that [Foundry](https://book.getfoundry.sh) (Forge) is correctly installed and up to date. You can update Foundry by running:
+
+```shell
+foundryup
+```
+
+Then, install the project dependencies using Forge:
+
+```shell
+forge install
+```
+
+## Prepare Environment
+
+To deploy contracts and interact with the data oracle, you need:
+
+1. A wallet with gas tokens on **Arbitrum Sepolia**
+2. An OpenAI API key
+3. A Trust Domain (TD) address configured with your OpenAI API key
```

Comment:
How can a user get one? Is it one available at https://docs.quex.tech/general-information/addresses#arbitrum-sepolia? Or we need a separate TD for every contract?